lib/database/sthDatabase.js

```diff
@@ -313,6 +313,60 @@ function getCollection(params, options, callback) {
         return process.nextTick(callback.bind(null, error));
     }
 
+    //Call fetchCollection after creating database connection if connection was not restored after database restarted
+    if (isConnectionAlive()) {
+        fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+    } else {
+        reconnectDB(callback, function() {
+            fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+        });
+    }
+}
+
+/**
+ * Check whether database is connected or not
+ */
+function isConnectionAlive() {
+    let dbConnected = true;
+    if (db === null || !client.isConnected()) {
```

Comment:
Maybe better: 
```
 if (db === null || (client && !client.isConnected()) {
```

Reply:
@AlvaroVega , Done is commit https://github.com/telefonicaid/fiware-sth-comet/pull/574/commits/305bd86a2578d6e406262cf509f49363d5261e95 below. Thanks :)